### PR TITLE
Add note about async and flesh out id and field property descriptions…

### PR DESCRIPTION
**Pull request by blackbaud-johnly**
… for grid component

### CODE REVIEW

File: src/app/components/grid/index.html [hunk 1]
```diff
@@ -62,13 +62,15 @@
   <sky-demo-page-properties sectionHeading="Grid column component properties">
     <sky-demo-page-property
       propertyName="id"
+      isOptional="true"
     >
-    Specifies a unique ID for the column. If no <stache-code>field</stache-code> property exists on the column, then this is used as the property to retreive cell information from an entry on the grid <stache-code>data</stache-code> array.
+    Specifies a unique ID for the column. You must provide either the <stache-code>id</stache-code> or <stache-code>field</stache-code> property for every column, but do not provide both. If <stache-code>field</stache-code> does not exist on a column, then the <stache-code>id</stache-code> property retreives cell information from an entry on the grid <stache-code>data</stache-code> array.
```
Comment (Blackbaud-SteveBrush) [inline src/app/components/grid/index.html]: "retreives"

File: src/app/components/grid/index.html [hunk 2]
```diff
@@ -62,13 +62,15 @@
...
     </sky-demo-page-property>
     <sky-demo-page-property
       propertyName="field"
+      isOptional="true"
     >
-    Specifies the property to retreive cell information from an entry on the grid <stache-code>data</stache-code> array. If no <stache-code>id</stache-code> property exists on the column, then this will be used as the entry for the grid <stache-code>selectedColumnIds</stache-code> array.
+    Specifies the property to retreive cell information from an entry on the grid <stache-code>data</stache-code> array. You must provide either the <stache-code>id</stache-code> or <stache-code>field</stache-code> property for every column, but do not provide both. If <stache-code>id</stache-code> does not exist on a column, then <stache-code>field</stache-code> is the entry for the grid <stache-code>selectedColumnIds</stache-code> array.
```
Comment (Blackbaud-SteveBrush) [inline src/app/components/grid/index.html]: "retreive"

File: src/app/learn/get-started/advanced/accessibility-unit-tests/index.html [hunk 3]
```diff
@@ -61,6 +61,14 @@
       expect(element).toBeAccessible();
     }));
   </stache-code-block>
+
+  <p>
+    Keep in mind that <stache-code>async</stache-code> is also a native JavaScript label, so you need to explicitly import the Angular <stache-code>async</stache-code> method at the top of the unit test spec file.
```
Comment (Blackbaud-SteveBrush) [inline src/app/learn/get-started/advanced/accessibility-unit-tests/index.html]: I would say "native JavaScript declaration,". I know I said "label" on the phone; that wasn't precise enough.

**Comment by blackbaud-johnly** [comment]
Addresses https://github.com/blackbaud/skyux2/issues/1767 and https://github.com/blackbaud/skyux2/issues/1768

**Pull request merged**